### PR TITLE
Adding Ephys Upload and Script for reconversions

### PR DESCRIPTION
**Pull request by Ahad-Allen**
This PR is intended to be used to allow for the regeneration and upload of ephys sessions. It also implements the updated upload process from the Ophys Pipeline, allowing for more accurate subject information to be uploaded.

### CODE REVIEW

File: scripts/dandi_ephys_uploads.py [hunk 1]
```diff
@@ -22,50 +40,105 @@ def get_creds():
 
 
 def set_env():
+    """Sets the environment variable for the DANDI API key
```
Comment (jeromelecoq) [inline scripts/dandi_ephys_uploads.py:43]: Is this function necessary? Are you calling it multiple times? 
It takes no input and return no output. 
I am concerned this add unnecessary lines to the codebase.

File: scripts/dandi_ephys_uploads.py [hunk 2]
```diff
@@ -22,50 +40,105 @@ def get_creds():
...
+
+    # Rename the nwb file to include the session id
+    src = os.path.join(nwb_folder_path,  "spike_times.nwb")
+    dst = os.path.join(directory_path, session_id + ".nwb")
+
+    # Move the nwb file to the session directory
+    shutil.move(src, dst)
+    
+
+    dandi_download(urls=r'https://dandiarchive.org/dandiset/' + dandiset_id + '/draft/', output_dir=str(dandi_path), get_metadata=True, get_assets=False)
```
Comment (jeromelecoq) [inline scripts/dandi_ephys_uploads.py:119]: So this download the entire dandiest, correct?

Comment (jeromelecoq) [reply scripts/dandi_ephys_uploads.py:119]: Will this force each upload of one session trigger the download of the entire dataset?

File: scripts/dandi_ephys_uploads.py [hunk 3]
```diff
@@ -22,50 +40,105 @@ def get_creds():
...
+    dandi_organize(paths=str(directory_path), dandiset_path=str(dandi_path_set))
+
+    # We need to add ses to file names manually when using dandi_organize
+    organized_nwbfiles = glob(join(directory_path, dandiset_id,  'sub-*' ,  "*.nwb" ))
+    for organized_nwbfile in organized_nwbfiles:
+        file_path = Path(organized_nwbfile)
+        if "ses" not in file_path.stem:
```
Comment (jeromelecoq) [inline scripts/dandi_ephys_uploads.py:126]: Wouldn't dandi organize use the session id directly if the nwb file had this string be the core object_id of the dataset?

File: src/openscopenwb/utils/firebase_functions.py [hunk 4]
```diff
@@ -207,9 +208,19 @@ def update_o_session(project_id, session_id):
...
+        status = view_session(project_id, session_id)['status']
+        notes = view_session(project_id, session_id)['notes']
+        #allen_dir = view_session(project_id, session_id)['path']
+        dandi = view_session(project_id, session_id)['dandi']
+        meta_dict['status'] = status
+        meta_dict['notes'] = notes
+        #meta_dict['path'] = allen_dir
+        meta_dict['dandi'] = dandi    
+        ref.update(meta_dict)
+    except KeyError:
```
Comment (jeromelecoq) [inline src/openscopenwb/utils/firebase_functions.py:222]: Why do we need a try / except here? Is this part of the normal code run?

File: scripts/dandi_ephys_uploads.py (hunk 2 above)
Comment (Ahad-Allen) [reply scripts/dandi_ephys_uploads.py:119]: This only downloads the metadata asssociated with the set, not the assets. So just the dandiset.yaml

**Comment by Ahad-Allen** [comment]
I think the idea of creating the upload test is good!

**Comment by jeromelecoq** [comment]
this looks good. Thanks for adding the tests!

**Pull request merged**